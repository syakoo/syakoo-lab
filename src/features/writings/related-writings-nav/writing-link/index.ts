export { WritingLink } from "./writing-link";
