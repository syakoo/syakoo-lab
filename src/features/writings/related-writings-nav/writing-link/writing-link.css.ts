import { style } from "@vanilla-extract/css";

import { theme } from "@/design-system/theme.css";

export const writingLinkStyles = {
  iconWrapper: style({
    width: theme.fontSize[100],
  }),
};
