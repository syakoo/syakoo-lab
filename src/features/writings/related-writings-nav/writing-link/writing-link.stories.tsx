import type { Meta, StoryObj } from "@storybook/react";

import { WritingLink } from ".";

const meta = {
  component: WritingLink,
  tags: ["autodocs"],
  parameters: { testLevel: "snapshot" },
} satisfies Meta<typeof WritingLink>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Article: Story = {
  args: {
    head: {
      id: "sample-writing-meta-id",
      title: "Sample Article",
      type: "article",
      tags: ["雑記", "Sample"],
      published: "2023-10-09",
    },
  },
};

export const Note: Story = {
  args: {
    head: {
      id: "sample-writing-meta-id",
      title: "Sample Note",
      type: "note",
      tags: ["雑記", "Sample"],
      published: "2023-10-09",
    },
  },
};

export const Sample: Story = {
  args: {
    head: {
      id: "sample-writing-meta-id",
      title: "Sample Diary",
      type: "diary",
      tags: ["雑記", "Sample"],
      published: "2023-10-09",
    },
  },
};
