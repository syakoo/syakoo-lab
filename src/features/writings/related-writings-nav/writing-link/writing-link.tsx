import type { WritingHead } from "@/entities/writing/models/writing";
import { writingPaths } from "@/entities/writing/paths/writing-paths";
import { writingTypeConfig } from "@/entities/writing/writing-type";
import { Icon } from "@/shared/design-system/icons";
import { Col, Row } from "@/shared/design-system/layout";
import { Link } from "@/shared/design-system/ui";
import { H4, Span } from "@/shared/design-system/ui/text/text";

import { writingLinkStyles } from "./writing-link.css";

type WritingLinkProps = {
  head: WritingHead;
};

export const WritingLink: React.FC<WritingLinkProps> = ({ head }) => {
  const { iconName } = writingTypeConfig.find(
    ({ type }) => type === head.type,
  )!;

  return (
    <Row align="flexStart" as="article" gap="100">
      <div className={writingLinkStyles.iconWrapper}>
        <Icon name={iconName} />
      </div>
      <Col gap="50">
        <Link href={writingPaths.detail(head.id)}>
          <H4 size="100">{head.title}</H4>
        </Link>
        <Span color="secondary" size="50">
          {head.published}
        </Span>
      </Col>
    </Row>
  );
};
