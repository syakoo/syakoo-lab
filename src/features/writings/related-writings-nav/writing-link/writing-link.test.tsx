import { testStories } from "@/test/test-stories";

import * as stories from "./writing-link.stories";

describe("WritingLink", () => {
  testStories(stories);
});
