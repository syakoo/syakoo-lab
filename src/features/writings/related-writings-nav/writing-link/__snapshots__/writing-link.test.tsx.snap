// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WritingLink スナップショット Article 1`] = `
<body>
  <div>
    <article
      class="flex_flexStyle__12xne02i flex_flexStyle_align_flexStart__12xne02r flex_flexStyle_variants_align_flexStart__12xne026 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
    >
      <div
        class="writing-link_writingLinkStyles_iconWrapper__txjmpc0"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 14.25v-2.625a3.375 3.375 0 00-3.375-3.375h-1.5A1.125 1.125 0 0113.5 7.125v-1.5a3.375 3.375 0 00-3.375-3.375H8.25m0 12.75h7.5m-7.5 3H12M10.5 2.25H5.625c-.621 0-1.125.504-1.125 1.125v17.25c0 .621.504 1.125 1.125 1.125h12.75c.621 0 1.125-.504 1.125-1.125V11.25a9 9 0 00-9-9z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_column__12xne02k flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
      >
        <a
          class="link_linkStyle__jc9bkl0"
          href="/writings/sample-writing-meta-id"
        >
          <h4
            class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_100__xbx8xul text_textStyle_variants_size_100__xbx8xu6 text_textStyle_weight_bold__xbx8xut"
          >
            Sample Article
          </h4>
        </a>
        <span
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_50__xbx8xuj text_textStyle_variants_size_50__xbx8xu4 text_textStyle_weight_normal__xbx8xuu"
        >
          2023-10-09
        </span>
      </div>
    </article>
  </div>
</body>
`;

exports[`WritingLink スナップショット Note 1`] = `
<body>
  <div>
    <article
      class="flex_flexStyle__12xne02i flex_flexStyle_align_flexStart__12xne02r flex_flexStyle_variants_align_flexStart__12xne026 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
    >
      <div
        class="writing-link_writingLinkStyles_iconWrapper__txjmpc0"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L10.582 16.07a4.5 4.5 0 01-1.897 1.13L6 18l.8-2.685a4.5 4.5 0 011.13-1.897l8.932-8.931zm0 0L19.5 7.125M18 14v4.75A2.25 2.25 0 0115.75 21H5.25A2.25 2.25 0 013 18.75V8.25A2.25 2.25 0 015.25 6H10"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_column__12xne02k flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
      >
        <a
          class="link_linkStyle__jc9bkl0"
          href="/writings/sample-writing-meta-id"
        >
          <h4
            class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_100__xbx8xul text_textStyle_variants_size_100__xbx8xu6 text_textStyle_weight_bold__xbx8xut"
          >
            Sample Note
          </h4>
        </a>
        <span
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_50__xbx8xuj text_textStyle_variants_size_50__xbx8xu4 text_textStyle_weight_normal__xbx8xuu"
        >
          2023-10-09
        </span>
      </div>
    </article>
  </div>
</body>
`;

exports[`WritingLink スナップショット Sample 1`] = `
<body>
  <div>
    <article
      class="flex_flexStyle__12xne02i flex_flexStyle_align_flexStart__12xne02r flex_flexStyle_variants_align_flexStart__12xne026 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
    >
      <div
        class="writing-link_writingLinkStyles_iconWrapper__txjmpc0"
      >
        <svg
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6.042A8.967 8.967 0 006 3.75c-1.052 0-2.062.18-3 .512v14.25A8.987 8.987 0 016 18c2.305 0 4.408.867 6 2.292m0-14.25a8.966 8.966 0 016-2.292c1.052 0 2.062.18 3 .512v14.25A8.987 8.987 0 0018 18a8.967 8.967 0 00-6 2.292m0-14.25v14.25"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
      <div
        class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_column__12xne02k flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
      >
        <a
          class="link_linkStyle__jc9bkl0"
          href="/writings/sample-writing-meta-id"
        >
          <h4
            class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_100__xbx8xul text_textStyle_variants_size_100__xbx8xu6 text_textStyle_weight_bold__xbx8xut"
          >
            Sample Diary
          </h4>
        </a>
        <span
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_50__xbx8xuj text_textStyle_variants_size_50__xbx8xu4 text_textStyle_weight_normal__xbx8xuu"
        >
          2023-10-09
        </span>
      </div>
    </article>
  </div>
</body>
`;
